tilings/strategies/assumption_insertion.py

```diff
@@ -181,3 +198,51 @@ def to_jsonable(self) -> dict:
     @classmethod
     def from_dict(cls, d: dict) -> "AddAssumptionFactory":
         return cls()
+
+
+class AddInterleavingAssumptionFactory(StrategyFactory[Tiling]):
+    def __init__(self, unions: bool = True):
+        self.unions = unions
+
+    @staticmethod
+    def strategy_from_components(
+        comb_class: Tiling, components: Tuple[Tuple[Cell, ...], ...]
+    ) -> Iterator[Rule]:
+        """
+        Yield an AddAssumption strategy for the given component if needed.
+        """
+        cols, rows = interleaving_rows_and_cols(components)
+        assumptions = chain.from_iterable(
+            assumptions_to_add(cells, cols, rows) for cells in components
+        )
+        if assumptions:
+            strategy = AddAssumptionsStrategy(assumptions, workable=True)
+            yield strategy(comb_class)
+
+    # TODO: unions? monotone?
```

Comment:
What about this todo? I think it's doing unions but it's not doing monotone

Reply:
Are you adding this when you add interleaving splitting?

Reply:
unions are done, monotone not. Left this as a todo- currently there is no purpose in using monotone. When it can do something more than interleaving in terms of enumeration we can come back and think about this. At the very least, using unions will certainly suffice when using monotone interleaving